src/lex.cpp

```diff
@@ -634,34 +657,55 @@ void lexer::parse_binary_number() {
   this->input_ = check_garbage_in_number_literal(input);
 }
 
-const char8* lexer::check_garbage_in_number_literal(const char8* input) {
+void lexer::parse_octal_number(bool strict) {
+  const char8* input = this->input_;
+
+  while (this->is_octal_digit(*input)) {
+    ++input;
+  }
+
   const char8* garbage_begin = input;
   for (;;) {
     switch (*input) {
+    /* TODO (👮🏾‍♀️ mswitch switch to just 8 and 9)*/
+    // decimal point should also be included
     QLJS_CASE_DECIMAL_DIGIT:
+      if (strict) {
+        ++input;
+        goto done_parsing_garbage;
+      } else {
+        this->input_ = input;
+        this->parse_number();
+        return;
+      }
     QLJS_CASE_IDENTIFIER_START:
-      input += 1;
-      break;
+    case u8'.':
+      ++input;
+      goto done_parsing_garbage;
```

Comment:
Maybe this isn't the right thing.  But the issue was if I had 

```suggestion
      break;
```

a decimal like `012.3` would see the `.`, break, increment, then the `3` would be seen by [678](https://github.com/strager/quick-lint-js/pull/40/commits/d44727a46e5cca624bcc21f421ca05c5bb331571#diff-3f07da05c7d34d6ae43baceb4a820032R684) and the execution disappears forever.

Reply:
May be resolved as of bf6e92bb6a2fa9087392459c181032234b210985